core/domain/takeout_service.py

```diff
@@ -42,34 +75,25 @@ def export_data_for_user(user_id):
                                 model export policy>
         }
     """
-    models_to_export = [
-        user_models.UserStatsModel,
-        user_models.UserSettingsModel,
-        user_models.UserSubscriptionsModel,
-        user_models.UserSkillMasteryModel,
-        user_models.UserContributionsModel,
-        user_models.ExplorationUserDataModel,
-        user_models.CompletedActivitiesModel,
-        user_models.IncompleteActivitiesModel,
-        user_models.ExpUserLastPlaythroughModel,
-        user_models.LearnerPlaylistModel,
-        user_models.CollectionProgressModel,
-        user_models.StoryProgressModel,
-        feedback_models.GeneralFeedbackThreadModel,
-        feedback_models.GeneralFeedbackMessageModel,
-        collection_models.CollectionRightsModel,
-        suggestion_models.GeneralSuggestionModel,
-        exploration_models.ExplorationRightsModel,
-        email_models.GeneralFeedbackEmailReplyToIdModel
+    all_models = []
+    model_names_list = [
+        model_name for model_name in dir(models.NAMES)
+        if not model_name.startswith('__') and not model_name == 'base_model'
     ]
+    model_modules = models.Registry.import_models(model_names_list)
+    for model_module in model_modules:
+        for _, obj in inspect.getmembers(model_module):
+            if inspect.isclass(obj):
+                all_models.append(obj)
 
     exported_data = dict()
-    for model in models_to_export:
+    models_to_export = get_models_should_be_exported()
+    for model in all_models:
         # Split the model name by uppercase characters.
```

Comment:
Any reason why models_to_export doesn't just contain strings that correspond to `model.__name__`?

Reply:
No longer relevant in most recent push.